arch/src/x86_64/mod.rs

```diff
@@ -130,6 +130,7 @@ pub fn configure_system(
     num_cpus: u8,
     setup_hdr: Option<setup_header>,
     _serial_enabled: bool,
+    _end_of_range: GuestAddress,
```

Comment:
That's actually nice to provide the end of range, and I would suggest that we use it to replace every occurrence of `guest_mem.end_addr()`. This will remove the assumption that `GuestMemoryMmap` is only about RAM.

Reply:
But those two are not the the same. `end_of_range` is the last byte in the entire address space.  `guest_mem.end_addr()` is the size of the RAM minus 1. Or it is at least at the point it is used to calculate the E820 tables in `configure_system()`. 

Your point is still valid though, we should look at having a different, explicit parameter to store the `end_of_ram` in particular if we start to use `GuestMemoryMmap` for handing the currently manual mmap()s then it will go wrong.

Reply:
I opened #289 to track this.

Reply:
Great thanks!